presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -159,7 +167,9 @@ private Type instantiateParametricType(TypeSignature signature)
             return null;
         }
         Type instantiatedType = parametricType.createType(parameters);
-        checkState(instantiatedType.getTypeSignature().equals(signature), "Instantiated parametric type name (%s) does not match expected name (%s)", instantiatedType, signature);
+
+        // TODO: reimplement this check? Currently "varchar(Integer.MAX_VALUE)" fails with "varchar"
```

Comment:
Put explicit precondition here that we expect `expectedTypeParameters.size()` to be either 0 or equal to size of `actualTypeParameters.size()`.